src/test/compile-fail/feature-gate-static_recursion.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+static mut S: *const u8 = unsafe { &S as *const *const u8 as *const u8 };
+//~^ ERROR recursive static (see issue #29719)
+
+struct StaticDoubleLinked {
+    prev: &'static StaticDoubleLinked,
+    next: &'static StaticDoubleLinked,
+    data: i32,
+    head: bool,
+}
+
+static L1: StaticDoubleLinked = StaticDoubleLinked{prev: &L3, next: &L2, data: 1, head: true};
+//~^ ERROR recursive static (see issue #29719)
+//~^^ ERROR recursive static (see issue #29719)
+//~^^^ ERROR recursive static (see issue #29719)
```

Comment:
```static L1: StaticDoubleLinked = StaticDoubleLinked{prev: &L3, next: &L2, data: 1, head: true};
//~^ ERROR recursive static (see issue #29719)
static L2: StaticDoubleLinked = StaticDoubleLinked{prev: &L1, next: &L3, data: 2, head: false};
//~^ ERROR recursive static (see issue #29719)
static L3: StaticDoubleLinked = StaticDoubleLinked{prev: &L2, next: &L1, data: 3, head: false};
//~^ ERROR recursive static (see issue #29719)
```

fails with this error:
```unexpected errors (from JSON output): [
    Error {
        line_num: 21,
        kind: Some(
            Error
        ),
        msg: "21:1: 21:95: recursive static (see issue #29719)"
    },
    Error {
        line_num: 21,
        kind: Some(
            Error
        ),
        msg: "21:1: 21:95: recursive static (see issue #29719)"
    }
]

not found errors (from test file): [
    Error {
        line_num: 23,
        kind: Some(
            Error
        ),
        msg: "recursive static (see issue #29719)"
    },
    Error {
        line_num: 25,
        kind: Some(
            Error
        ),
        msg: "recursive static (see issue #29719)"
    }
]```

Sorry, I probably should have led with this in the comments as opposed to making a dumb looking edit